scalameta/semanticdb-scalac/src/main/scala/scala/meta/internal/semanticdb/DocumentOps.scala

```diff
@@ -153,7 +153,27 @@ trait DocumentOps { self: DatabaseOps =>
                 // https://github.com/scalameta/scalameta/issues/665
                 // Instead of crashing with "unsupported file", we ignore these cases.
                 if (mtree.pos == m.Position.None) return
-                if (names.contains(mtree.pos)) return // NOTE: in the future, we may decide to preempt preexisting db entries
+
+                object DesugaredLoopMethod {
+                  def unapply(sym: m.Symbol): Option[String] = sym match {
+                    case m.Symbol.Global(_, m.Signature.Method(name, _)) => Some(name)
+                    case _ => None
+                  }
+                }
```

Comment:
I'd move this extractor to SymbolOps and would rename it to something like `MethodSymbol`, since it works for all methods, not just for desugared loop methods.

Reply:
Good point. Done.